thirdparty/runtime_environments.go

```diff
@@ -96,3 +96,56 @@ func (c *RuntimeEnvironmentsClient) getOSInfo(ctx context.Context, amiID string,
 	}
 	return osInfo, nil
 }
+
+// Toolchain represents a toolchain's information.
+type Toolchain struct {
+	Name    string
+	Version string
+	Manager string
+}
+
+// ToolchainFilterOptions represents the filtering arguments, each of which is optional except for the AMIID.
+type ToolchainFilterOptions struct {
+	AMIID   string
+	Page    int
+	Limit   int
+	Name    string // Filter by the name of the toolchain (ex. golang).
+	Version string // Filter by the version (ex. go1.8.7).
+}
+
+// getPackages returns a list of packages from the corresponding AMI id.
+func (c *RuntimeEnvironmentsClient) getToolchains(ctx context.Context, opts ToolchainFilterOptions) ([]Toolchain, error) {
+	params := url.Values{}
+	if opts.AMIID == "" {
+		return nil, errors.New("no AMI provided")
+	}
```

Comment:
nit. swap the order; if an error happens, then we would have defined a variable that doesn't get used
```suggestion
	if opts.AMIID == "" {
		return nil, errors.New("no AMI provided")
	}
	params := url.Values{}
```